doc/case-studies/xycloans/MCxycloans_monitor.ejs.tla

```diff
@@ -0,0 +1,167 @@
+------------------------------- MODULE MC -------------------------------
+<%#
+/*
+ * An EJS template for generating the initial state from the aggregated state of the contract.
+ *
+ * Usage:
+ *
+ * npx ejs MCxycloans_monitor.ejs.tla -f state.json >MC.tla
+ *
+ * Igor Konnov, 2024
+ */
+%>
+(* THIS MODULE IS AUTOGENERATED FROM SOROBAN STATE *)
+EXTENDS Integers, Apalache, xycloans_types
+
+\* the set of all possible token amounts
+AMOUNTS == Nat
+\* the contract address for the xycLoans contract
+XYCLOANS == "<%- contractId %>"
+\* the token address
+XLM_TOKEN_SAC_TESTNET == "<%- storage[contractId].instance.TokenId %>"
+
+\* user-controlled addresses
+USER_ADDR == {
+    <%
+    const balanceAddrs =
+        Object.keys(storage[contractId].persistent)
+            .filter((key) => key.startsWith("Balance,"))
+            .map((key) => key.split(",")[1])
+    %>
+    <%-
+    balanceAddrs
+        .map((addr) => `    "${addr}"`)
+        .join(",\n    ")
```

Comment:
Bit confused about the whitespace here. Looks like you're indenting twice?

Reply:
oh, that's true. Copilot is doing weird things sometimes :)